src/http/http_rpc_context.h

```diff
@@ -303,6 +303,14 @@ namespace http
       return status_success(response_status);
     }
 
+    virtual void reset_response() override
```

Comment:
It would be nice to have unit tests that construct two contexts, run a request + reset_response on one, and perform a structural equality check to make sure we're not missing anything here if&when we add state in the future.